lib/datasource/terraform-provider/index.ts

```diff
@@ -54,6 +56,11 @@ async function queryRegistry(
   dep.releases = res.versions.map((version) => ({
     version,
   }));
+  // set published date for latest release
+  const currentVersionIndex = dep.releases.findIndex((release) => {
+    return res.version === release.version;
+  });
+  dep.releases[currentVersionIndex].releaseTimestamp = res.published_at;
```

Comment:
```suggestion
  const currentVersion = dep.releases.find((release) => {
    return res.version === release.version;
  });
  if (currentVersion) {
    currentVersion.releaseTimestamp = res.published_at;
  }
```
use find and add null check